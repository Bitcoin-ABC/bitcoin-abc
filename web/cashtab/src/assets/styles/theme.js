export const theme = {
    primary: '#00ABE7',
    brandSecondary: '#CD0BC3',
    contrast: '#fff',
    app: {
        sidebars: `url("/cashtab_bg.png")`,
        background: '#fbfbfd',
    },
    wallet: {
        background: '#fff',
        text: {
            primary: '#273498',
            secondary: '#273498',
        },
        switch: {
            activeCash: {
                shadow:
                    'inset 8px 8px 16px #0074C2, inset -8px -8px 16px #273498',
            },
            activeToken: {
                background: '#CD0BC3',
                shadow:
                    'inset 5px 5px 11px #FF21D0, inset -5px -5px 11px #CD0BC3',
            },
            inactive: {
                background: 'linear-gradient(145deg, #eeeeee, #c8c8c8)',
            },
        },
        borders: { color: '#e2e2e2' },
        shadow: 'rgba(0, 0, 0, 1)',
    },
    tokenListItem: {
        background: '#ffffff',
        color: '',
        boxShadow:
            'rgba(0, 0, 0, 0.01) 0px 0px 1px, rgba(0, 0, 0, 0.04) 0px 4px 8px,rgba(0, 0, 0, 0.04) 0px 16px 24px, rgba(0, 0, 0, 0.01) 0px 24px 32px',
        border: '#e9eaed',
        hoverBorder: '#231F20',
    },
    footer: {
        background: '#fff',
        navIconInactive: '#949494',
    },
    forms: {
        error: '#FF21D0',
        border: '#e7edf3',
        text: '#001137',
        addonBackground: '#f4f4f4',
        addonForeground: '#3e3f42',
        selectionBackground: '#fff',
    },
    icons: { outlined: '#273498' },
    modals: {
        buttons: { background: '#fff' },
    },
    settings: { delete: '#CD0BC3' },
    qr: {
        copyBorderCash: '#00ABE7',
        copyBorderToken: '#FF21D0',
        background: '#fff',
        token: '#231F20',
        shadow:
            'rgba(0, 0, 0, 0.01) 0px 0px 1px, rgba(0, 0, 0, 0.04) 0px 4px 8px, rgba(0, 0, 0, 0.04) 0px 16px 24px, rgba(0, 0, 0, 0.01) 0px 24px 32px',
    },
    buttons: {
        primary: {
            backgroundImage:
                'linear-gradient(270deg, #0074C2 0%, #273498 100%)',
            color: '#fff',
            hoverShadow: '0px 3px 10px -5px rgba(0, 0, 0, 0.75)',
        },
        secondary: {
            background: '#e9eaed',
            color: '#444',
            hoverShadow: '0px 3px 10px -5px rgba(0, 0, 0, 0.75)',
        },
    },
    collapses: {
        background: '#fbfcfd',
        border: '#eaedf3',
        color: '#3e3f42',
    },
};
