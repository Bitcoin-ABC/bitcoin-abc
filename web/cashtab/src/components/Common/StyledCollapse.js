import styled from 'styled-components';
import { Collapse } from 'antd';

export const StyledCollapse = styled(Collapse)`
    background: ${props => props.theme.collapses.background} !important;
    border: 1px solid ${props => props.theme.collapses.border} !important;

    .ant-collapse-content {
        border: 1px solid ${props => props.theme.collapses.border};
        border-top: none;
    }

    .ant-collapse-item {
        border-bottom: none !important;
    }

    *:not(button) {
        color: ${props => props.theme.collapses.color} !important;
    }
`;

export const TokenCollapse = styled(Collapse)`
    ${({ disabled = false, ...props }) =>
        disabled === true
            ? `
                background: ${props.theme.buttons.secondary.background} !important;
           .ant-collapse-header {
               font-size: 18px;
               font-weight: bold;
               color: ${props.theme.buttons.secondary.color} !important;
               svg {
                   color: ${props.theme.buttons.secondary.color} !important;
               }
           }
           .ant-collapse-arrow {
               font-size: 18px;
           }
            `
            : `
                background: ${props.theme.primary} !important;
           .ant-collapse-header {
               font-size: 18px;
               font-weight: bold;
               color: ${props.theme.contrast} !important;
               svg {
                   color: ${props.theme.contrast} !important;
               }
           }
           .ant-collapse-arrow {
               font-size: 18px;
           }
            `}
`;
