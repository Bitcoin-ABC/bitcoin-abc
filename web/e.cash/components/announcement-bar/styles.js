// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';

export const AnnouncementBarCtn = styled.div`
    a,
    div {
        --barHeight: 44px;
        width: 100%;
        background-color: ${props => props.theme.colors.black};
        padding: 0px 24px;
        text-align: center;
        color: ${props => props.theme.colors.contrast};
        font-weight: 600;
        letter-spacing: 1px;
        text-transform: uppercase;
        font-size: 14px;
        height: var(--barHeight);
        display: flex;
        align-items: center;
        justify-content: center;
        transition: all 200ms ease-in-out;
        margin-top: ${props =>
            props.navBackground ? `calc(var(--barHeight) * -1)` : '0'};
    }
    a {
        :hover {
            background-color: ${props => props.theme.colors.accent};
            color: ${props => props.theme.colors.contrast};
        }
    }

    ${props => props.theme.breakpoint.medium} {
        a,
        div {
            padding: 0px 10px;
            font-size: 12px;
            --barHeight: 40px;
        }
    }

    a > span {
        display: flex;
        flex-direction: column;
        gap: 2px;
    }

    a > span span {
        line-height: 1em;
    }
`;

export const Timer = styled.span`
    color: ${props => props.theme.colors.primaryLight};
`;
