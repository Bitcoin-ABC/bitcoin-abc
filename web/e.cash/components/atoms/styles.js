// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    max-width: ${props => (props.narrow ? '900px' : '1500px')};
    margin: auto;
    padding: 0 50px;

    ${props => props.theme.breakpoint.medium} {
        padding: 0 20px;
    }
`;

export const ThemeSwitch = styled.div`
    position: fixed;
    bottom: 30px;
    width: 40px;
    height: 40px;
    z-index: 9999;
    background: rgba(255, 255, 255, 0.1);
    right: 30px;
`;

export const CenterImage = styled.div`
    position: relative;
    width: 100%;
    height: ${props => props.height};
    ${props => props.theme.filters.grayscale}

    img {
        object-fit: contain;
    }

    ${props => props.theme.breakpoint.medium} {
        height: calc(${props => props.height} / 2);
    }
`;

export const GradientSpacer = styled.div`
    height: 100px;
    width: 100%;
    background-image: linear-gradient(
        180deg,
        ${props => props.theme.colors.darkBlue},
        ${props => props.theme.colors.darkBackground}
    );
    margin-bottom: 80px;
`;

export const WarningBox = styled.p`
    padding: 20px;
    background-color: ${props => props.theme.colors.primary};
    border-radius: 10px;
    line-height: 1.5em;
`;
