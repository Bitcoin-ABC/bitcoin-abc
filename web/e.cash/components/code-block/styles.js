// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';

export const Pre = styled.pre`
    background: ${props => props.theme.colors.black};
    width: 100%;
    overflow-x: scroll;
    padding: 20px;
`;
