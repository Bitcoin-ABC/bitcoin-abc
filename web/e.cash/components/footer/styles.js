// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';
import Link from 'next/link';

export const FooterCtn = styled.div`
    width: 100%;
    height: auto;
    padding: 40px 40px 20px;
    background-color: ${props => props.theme.colors.black};
    color: #fff;
    clip-path: polygon(
        0 0,
        18% 0,
        19% 20px,
        81% 20px,
        82% 0,
        100% 0,
        100% 100%,
        0 100%
    );
    margin-top: -20px;
    ${props => props.theme.filters.grayscale};

    ${props => props.theme.breakpoint.medium} {
        padding: 40px 0 20px;
    }
`;

export const LogoCtn = styled.div`
    width: 100%;
    border-bottom: 1px solid rgba(255, 255, 255, 0.3);
    margin-bottom: 30px;
    > div {
        width: 130px;
        height: 100px;
        position: relative;

        ${props => props.theme.breakpoint.medium} {
            width: 100px;
        }
    }
`;

export const ContentCtn = styled.div`
    display: flex;
    justify-content: space-between;

    ${props => props.theme.breakpoint.medium} {
        flex-direction: column;
    }
`;

export const NavItem = styled.div`
    font-size: 14px;
    font-weight: 500;
    color: #fff;
    margin-bottom: 5px;
`;

export const SocialCtn = styled.div`
    display: flex;
    width: 400px;
    height: 30px;
    margin-top: 10px;

    ${props => props.theme.breakpoint.medium} {
        width: 100%;
        flex-wrap: wrap;
        height: auto;
    }

    a {
        width: 20px;
        height: 20px;
        position: relative;
        margin-right: 20px;
        transition: all ease-in-out 200ms;

        :hover {
            transform: scale(1.4);
        }

        :hover img {
            filter: invert(32%) sepia(76%) saturate(5535%) hue-rotate(189deg)
                brightness(93%) contrast(101%);
        }

        ${props => props.theme.breakpoint.medium} {
            margin-bottom: 15px;
            margin-right: 15px;
        }
    }
`;

export const ContactLink = styled(Link)`
    opacity: 0.5;
    font-size: 14px;
    color: #fff;
    line-height: 1.9em;

    :hover {
        opacity: 1;
        color: var(--bluelight);
    }
`;

export const LinksCtn = styled.div`
    display: flex;

    ${props => props.theme.breakpoint.medium} {
        flex-direction: column;
        margin-top: 50px;
    }

    > div {
        padding: 0;
        margin-left: 60px;
        ${props => props.theme.breakpoint.medium} {
            margin-left: 0;
            margin-bottom: 20px;
        }

        > div:first-child {
            font-size: 14px;
            font-weight: 500;
            color: #fff;
            margin-bottom: 5px;
        }
    }
`;

export const DropdownCtn = styled.div`
    display: flex;
    flex-direction: column;

    a {
        opacity: 0.5;
        font-size: 14px;
        color: #fff;
        line-height: 1.9em;

        :hover {
            opacity: 1;
            color: ${props => props.theme.colors.primaryLight};
        }
    }
`;

export const Copyright = styled.div`
    width: 100%;
    font-size: 14px;
    text-align: center;
    color: #fff;
    opacity: 0.3;
    margin: 50px 0 0px;
`;
