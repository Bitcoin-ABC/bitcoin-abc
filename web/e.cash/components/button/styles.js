// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';
import Link from 'next/link';
import { stealth } from '/styles/theme';

export const ButtonCtn = styled.div`
    ${props =>
        props.glow &&
        `filter: drop-shadow(0px 0px 10px ${
            props.color === 'accent'
                ? props.theme.colors.accent + '90'
                : props.color === 'white'
                ? props.theme.colors.contrast + '90'
                : props.theme.colors.primaryLight + '90'
        });`}
`;

export const ButtonMain = styled(Link)`
    display: inline-flex;
    padding: 2px;
    align-items: center;
    transition: all 300ms ease-in-out;
    color: ${props =>
        props.color === 'accent'
            ? props.theme.colors.accent
            : props.color === 'white'
            ? props.theme.colors.contrast
            : props.theme.colors.primaryLight};
    font-size: 15px;
    line-height: 1.5;
    font-weight: 400;
    letter-spacing: 1px;
    background-color: ${props =>
        props.color === 'accent'
            ? props.theme.colors.accent
            : props.color === 'white'
            ? props.theme.colors.contrast
            : props.theme.colors.primaryLight};
    ${props =>
        props.theme === stealth
            ? 'background-color: #fff !important; color: #fff !important; '
            : null};
    :hover {
        color: ${props =>
            props.color === 'accent'
                ? props.theme.colors.accent
                : props.color === 'white'
                ? props.theme.colors.contrast
                : props.theme.colors.primaryLight};
    }
`;

export const ButtonInner = styled.div`
    background-color: ${props => props.theme.colors.darkBlue};
    padding: 16px 40px;
    font-size: 15px;
    line-height: 1.5;
    font-weight: 500;
    letter-spacing: 1px;
    transition: all ease-in-out 150ms;

    &:hover {
        box-shadow: inset 0 0 13px 2px
            ${props =>
                props.color === 'accent'
                    ? props.theme.colors.accent + '90'
                    : props.color === 'white'
                    ? props.theme.colors.contrast + '90'
                    : props.theme.colors.primaryLight + '90'};
        animation: hoverbutton 150ms ease-in-out 1;
    }

    @keyframes hoverbutton {
        0% {
            background-color: ${props =>
                props.color === 'accent'
                    ? props.theme.colors.accent
                    : props.color === 'white'
                    ? props.theme.colors.contrast
                    : props.theme.colors.primaryLight} !important;
        }
        100% {
            background-color: unset;
        }
    }
    ${props => props.theme.breakpoint.medium} {
        .buttoninner {
            padding: 14px 30px;
            font-size: 14px;
        }
    }
`;
