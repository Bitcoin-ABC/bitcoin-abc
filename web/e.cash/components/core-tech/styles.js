// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.
import styled from 'styled-components';
import { motion } from 'framer-motion';
import { getAnimationSettings } from '/styles/framer-motion';

export const SectionCtn = styled(motion.div).attrs(() =>
    getAnimationSettings(),
)`
    position: relative;
    width: 100%;
    display: grid;
    grid-template-columns: 1fr auto 1fr;
    justify-content: center;
    margin-top: 60px;

    ${props => props.theme.breakpoint.medium} {
        grid-template-columns: 1fr;
        grid-template-rows: auto auto auto;
    }
`;

export const CoreTechImage = styled.div`
    width: 408px;
    height: 300px;
    position: relative;
    flex-shrink: 0;

    ${props => props.theme.breakpoint.medium} {
        display: none;
    }

    img {
        object-fit: contain;
    }
`;

export const Labels = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: space-between;
    height: 300px;

    ${props => props.theme.breakpoint.medium} {
        height: unset;
        margin-bottom: 5px;
    }

    & > :first-child {
        clip-path: polygon(20px 0, 100% 0, 100% 100%, 0 100%, 0 20px);
    }

    & > :first-child > div {
        clip-path: polygon(19px 0, 100% 0, 100% 100%, 0 100%, 0 19px);
    }

    & > :last-child {
        clip-path: polygon(
            100% 0,
            100% 100%,
            20px 100%,
            0 calc(100% - 20px),
            0 0
        );
    }

    & > :last-child > div {
        clip-path: polygon(
            100% 0,
            100% 100%,
            19px 100%,
            0 calc(100% - 19px),
            0 0
        );
    }

    ${props => props.theme.breakpoint.medium} {
        & > :first-child,
        & > :first-child > div,
        & > :last-child,
        & > :last-child > div {
            clip-path: none;
        }
    }
`;

export const LabelOuter = styled.div`
    padding: 2px;
    background-color: ${props => props.theme.colors.primaryLight};
    width: 100%;
    max-width: 400px;
    cursor: pointer;
    ${props => props.theme.breakpoint.medium} {
        max-width: unset;
        margin-bottom: 2px;
    }
`;

export const LabelInner = styled.div`
    background-color: ${props => props.theme.colors.darkBlue};
    font-weight: 600;
    font-size: 26px;
    padding: 5px 20px;
    text-align: right;

    ${props => props.theme.breakpoint.medium} {
        font-size: 20px;
        text-align: left;
    }

    div {
        width: 50px;
        height: 50px;
        position: relative;
        margin-bottom: 10px;

        img {
            object-fit: contain;
        }
    }
`;

export const DescriptionCtn = styled.div`
    display: flex;
    align-items: center;
    & > :first-child {
        padding: 3px;
        background-color: ${props => props.theme.colors.primaryLight};
        clip-path: polygon(
            0 0,
            calc(100% - 20px) 0%,
            100% calc(0% + 20px),
            100% 100%,
            0 100%
        );
        position: relative;
        ${props => props.theme.breakpoint.medium} {
            padding: 2px;
            clip-path: none;
        }
    }
    ${props => props.theme.breakpoint.medium} {
        grid-row: 2;
        grid-column: 1 / 3;
    }
`;

export const DescriptionInner = styled.div`
    background-color: ${props => props.theme.colors.darkBlue};
    clip-path: polygon(
        0 0,
        calc(100% - 19px) 0%,
        100% calc(0% + 19px),
        100% 100%,
        0 100%
    );
    font-size: 18px;
    padding: 20px 30px;
    text-align: left;
    ${props => props.theme.breakpoint.medium} {
        clip-path: none;
    }

    a {
        background-color: ${props => props.theme.colors.contrast};
        margin-top: 30px;
        display: inline-block;
        color: ${props => props.theme.colors.darkBlue};
        padding: 4px 20px;
        font-size: 14px;

        :hover {
            background-color: ${props => props.theme.colors.accent};
        }
    }

    div {
        width: 50px;
        height: 50px;
        position: relative;
        margin-bottom: 10px;

        img {
            object-fit: contain;
        }
    }
`;
