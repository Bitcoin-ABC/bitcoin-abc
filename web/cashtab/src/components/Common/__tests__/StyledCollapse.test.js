import React from 'react';
import renderer from 'react-test-renderer';
import { StyledCollapse } from '../StyledCollapse';
import { ThemeProvider } from 'styled-components';
import { theme } from '@assets/styles/theme';

test('Render StyledCollapse component', () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <StyledCollapse />
        </ThemeProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
