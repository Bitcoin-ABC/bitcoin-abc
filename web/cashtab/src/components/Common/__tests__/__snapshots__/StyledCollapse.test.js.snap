// Jest Snapshot v1, https://goo.gl/fbAQLP @generated

exports[`Render StyledCollapse component 1`] = `
<div
  className="ant-collapse ant-collapse-icon-position-left sc-bdVaJa AfOYf"
  role={null}
/>
`;
